src/crimson/os/seastore/lba_mapping.h

```diff
@@ -59,33 +59,14 @@ class LBAMapping {
     return parent->has_been_invalidated();
   }
 
-  bool is_unviewable_by_trans(CachedExtent& extent, Transaction &t) const {
-    if (!extent.is_valid()) {
-      return true;
-    }
-    if (extent.is_pending()) {
-      assert(extent.is_pending_in_trans(t.get_trans_id()));
-      return false;
-    }
-    auto &pendings = extent.mutation_pendings;
-    auto trans_id = t.get_trans_id();
-    bool unviewable = (pendings.find(trans_id, trans_spec_view_t::cmp_t()) !=
-		       pendings.end());
-    if (!unviewable) {
-      auto &trans = extent.retired_transactions;
-      unviewable = (trans.find(trans_id, trans_spec_view_t::cmp_t()) !=
-		 trans.end());
-      assert(unviewable == t.is_retired(extent.get_paddr(), extent.get_length()));
-    }
-    return unviewable;
-  }
-
   bool is_parent_viewable() const {
     ceph_assert(parent);
-    if (!parent->is_valid()) {
-      return false;
+    auto [viewable, state] = parent->is_viewable_by_trans(
+      ctx.trans.get_trans_id());
+    if (state == CachedExtent::viewable_state_t::stable_retired) {
+      assert(ctx.trans.is_retired(parent->get_paddr(), parent->get_length()));
```

Comment:
can it be moved to `is_viewable_by_trans()` as well ?

Reply:
Looks the transaction isn't available there, is it possible to keep this assert somewhere ?